### PR TITLE
Add /get-tgt end-point to bx509d

### PR DESCRIPTION
**Pull request by nicowilliams**

### CODE REVIEW

File: kdc/bx509d.8 [hunk 1]
```diff
@@ -127,23 +127,32 @@ required query parameter.
...
-.It Li dNSName = Va <hostname>
-.It Li rfc822Name = Va <email-address>
-.It Li xMPPName = Va <XMPP-address>
-.It Li krb5PrincipalName = Va <Kerberos-principal-name>
-.It Li ms-upn = Va <UPN>
-.It Li eku = Va <OID>
-.It Li lifetime = Va <lifetime>
+.It Li csr = Va base64-encoded-DER-encoded-CSR
+.It Li dNSName = Va hostname
+.It Li rfc822Name = Va email-address>
```
Comment (jheiss) [inline kdc/bx509d.8:150]: Should that trailing angle bracket go away?

Comment (nicowilliams) [reply kdc/bx509d.8:150]: Yes!  Thanks.

File: tests/kdc/check-bx509.in [hunk 2]
```diff
@@ -55,6 +55,8 @@ kdc="${kdc} --addresses=localhost -P $port"
 server=datan.test.h5l.se
 otherserver=other.test.h5l.se
 cache="FILE:${objdir}/cache.krb5"
+cachefile2="${objdir}/cache2.krb5"
```
Comment (nicowilliams) [inline tests/kdc/check-bx509.in:58]: I should make sure this gets removed with `make clean`.

File: tests/kdc/check-bx509.in [hunk 3]
```diff
@@ -429,6 +432,21 @@ trap "kill -9 ${kdcpid} ${bx509pid}; echo signal killing kdc and bx509d; exit 1;
...
+         --resolve ${server}:${bx509port}:127.0.0.1                     \
+         -H "Authorization: Negotiate $token"                           \
+         "http://${server}:${bx509port}/get-tgt"); then
+    ${klist2} ||
+        { echo "Failed to get a TGT with /get-tgt end-point (klist)"; exit 2; }
+else
+    echo "Failed to get a TGT with /get-tgt end-point"
+    exit 2
+fi
+
```
Comment (nicowilliams) [inline tests/kdc/check-bx509.in:449]: Maybe do more than `klist` the ticket, maybe also try to use it.

File: kdc/bx509d.c [hunk 4]
```diff
@@ -1138,30 +1127,70 @@ do_pkinit(struct bx509_request_desc *r)
...
+    if (ret == 0)
+        ret = krb5_get_init_creds_opt_alloc(r->context, &opt);
+    if (ret == 0)
         krb5_get_init_creds_opt_set_default_flags(r->context, "kinit", crealm,
                                                   opt);
-    if (ret == 0 &&
-        (ret = krb5_get_init_creds_opt_set_addressless(r->context,
-                                                       opt, 1)) == 0)
+    if (ret == 0)
+        ret = krb5_get_init_creds_opt_set_addressless(r->context, opt, 1);
```
Comment (nicowilliams) [inline kdc/bx509d.c:1192]: Here we're going to want to add support for specifying addresses in q-params.

**Pull request merged**